src/components/checkbox/__tests__/index.spec.tsx

```diff
@@ -24,57 +28,263 @@ describe('Checkbox renderer test', () => {
-    const component = <TestCase {...props}/>;
-    const driver = new CheckboxDriver({component, testID});
-
-    await driver.press();
-    expect(onValueChange).toHaveBeenCalledTimes(1);
-    expect(onValueChange).toHaveBeenCalledWith(true);
-    await driver.press();
-    expect(onValueChange.mock.calls).toEqual([[true], [false]]);
-    await driver.press();
-    expect(onValueChange.mock.calls).toEqual([[true], [false], [true]]);
+  describe('Press', () => {
+    it('Multiple clicks', async () => {
+      const props = {onValueChange};
+      const component = <TestCase {...props}/>;
+      const driver = new CheckboxDriver({component, testID});
+
+      await driver.press();
+      expect(onValueChange).toHaveBeenCalledTimes(1);
+      expect(onValueChange).toHaveBeenCalledWith(true);
+      await driver.press();
+      expect(onValueChange.mock.calls).toEqual([[true], [false]]);
+      await driver.press();
+      expect(onValueChange.mock.calls).toEqual([[true], [false], [true]]);
+    });
+
+    it.each`
+      checkboxInitialValue
+      ${false}
+      ${true}
+    `('Disabled (value = $checkboxInitialValue)', async ({checkboxInitialValue}: {checkboxInitialValue: boolean}) => {
+      const props = {onValueChange, value: checkboxInitialValue, disabled: true};
+      const component = <TestCase {...props}/>;
+      const driver = new CheckboxDriver({component, testID});
+
+      await driver.press();
+
+      expect(onValueChange).not.toHaveBeenCalled();
+    });
   });
 
-  it.each`
-  checkboxInitialValue
-  ${false}
-  ${true}
-  `('Disabled (value = $checkboxInitialValue)', async ({checkboxInitialValue}: {checkboxInitialValue: boolean}) => {
-    const props = {onValueChange, value: checkboxInitialValue, disabled: true};
-    const component = <TestCase {...props}/>;
-    const driver = new CheckboxDriver({component, testID});
+  describe('Validation', () => {
+    describe('onChangeValidity', () => {
+      it('should not been called', async () => {
+        const props = {onChangeValidity};
+        const component = <TestCase {...props}/>;
+        const driver = new CheckboxDriver({component, testID});
+        await driver.press();
+
+        expect(onChangeValidity).not.toHaveBeenCalled();
+      });
+      it('should been called with required prop', async () => {
+        const props = {onChangeValidity, required: true};
+        const component = <TestCase {...props}/>;
+        const driver = new CheckboxDriver({component, testID});
+        await driver.press();
+
+        expect(onChangeValidity).toHaveBeenCalled();
+        expect(onChangeValidity).toHaveBeenCalledWith(true);
+      });
+
+      it('should not been called after invoking validate()', async () => {
+        const props = {required: true, onChangeValidity};
+        const component = <TestCase {...props}/>;
+        //eslint-disable-next-line
+        new CheckboxDriver({component, testID});
+
+        checkboxRef.current?.validate();
```

Comment:
If you add a `?` you don't know that `validate` was actually called (what if `current` is `undefined`?